heron/spi/src/java/com/twitter/heron/spi/utils/ShellUtils.java

```diff
@@ -315,4 +315,15 @@ public static boolean extractPackage(
     return ret == 0;
   }
 
+  // java 7 compatible version of String.join(" ", array), available in java 8
+  private static String toString(String[] array) {
```

Comment:
I would suggest `joinString` name. The `toString` may be confusing with `Object.toString`.